### ISSUE TITLE
Ability to dismiss records when done

### ISSUE THREAD

**Issue by CloVasco** [benetech/ServiceNet#308]
Include a way for records to disappear after users:
-import what they need and are done with it
-are not interested in partners' data

So that they can clean their homepage as they work through the list and don't feel overwhelmed by a long list of things that are not relevant to them (any more) 

Questions
-for whom - the user or the team?
-what happens to the record - disappear  from the landing page forever until there's an update from another partner (unless the record is searched)?  is archived somewhere to be retrieved by user? 
-need to create a new button - I am done with this record

**Comment by CloVasco** [comment]
@laurabenetech to add 3 visuals:
Nber 1- visual of link to see  non-matching records that have been dismissed
Nber 2-visual of page when click on Nber 1 to see records that have been dismissed with info (record has been dismissed by person on date. reason for it
Nber 3- feedback window - thanks for your feedback. confirm that you want to dismiss with link to provide reason (non mandatory!!)

**Comment by laurabenetech** [comment]
@CloVasco Here is the Feedback window with note feature. I was able to leverage a design from Exygy. 

![Mismatch confirm_with_note](https://user-images.githubusercontent.com/48697326/55647548-cd235f80-5792-11e9-9595-4df409f8a94c.png)


I'll attach the other visuals before our mtg today.

**Comment by CloVasco** [comment]
@laurabenetech I think we have talked a lot about it and we heard from pilot partners.
Please submit visuals and details for what this feature would look like, including:
-the I am done button - where, name, etc
-and the page where users would be able to see records again after they have been dismissed

**Comment by laurabenetech** [comment]
@CloVasco 
Done button design for Home page:

When user clicks Close (X icon), on the on side-by-side page, the record is dismissed. Hovering over the X with the mouse displays the tool tip: “Done with this record"

Note: For this release, this choice affects the organization level (not the single user level). For a future release, users can choose in Settings to have this be at the individual user or org level. 


<img width="635" alt="Home page Done" src="https://user-images.githubusercontent.com/48697326/57403041-92e02000-718d-11e9-8c5e-ec30c1c1562d.png">

**Comment by laurabenetech** [comment]
@CloVasco 
Done button for Side by side page:
When user clicks Close (X icon), on the on side-by-side page, the record is dismissed. Hovering over the X with the mouse displays the tool tip: “Done with this record”

See visual here for specifications on other minor changes.

For this release, this choice affects the organization level (not the single user level). For a future release, users can choose in Settings to have this be at the individual user or org level. 

<img width="680" alt="Side by side Done" src="https://user-images.githubusercontent.com/48697326/57403234-126def00-718e-11e9-82d4-017301044647.png">

**Comment by laurabenetech** [comment]
@CloVasco 
Note: I'm working now on the page and link for dismissed records will update here when I have a draft ready

**Comment by laurabenetech** [comment]
@CloVasco 
Here is the page that displays records users have dismissed as Done.

<img width="601" alt="Page for viewing Done records" src="https://user-images.githubusercontent.com/48697326/57413045-ac8e6100-71a7-11e9-9f70-418d9c74e77b.png">

**Comment by laurabenetech** [comment]
@CloVasco 
Question about placement of link for dismissed records page:

It’s a bit tricky to find a place for the link to this page. The home page is an infinite scroll so we can’t put at bottom of that page, and the link to viewing removed records (bad match ones) is at the bottom of the side-by-side page (in the bad match area), so it could be confusing to have a link to dismissed records near that other link. 

Since this link most likely won’t be used often, it probably makes sense not to put it in a prominent place on the page (i.e., right at the top). 

Before I design the visual, I’m suggesting putting it in the drop down for the user, with the menu choice ““Display records dismissed as Done.” 

Let me know if that's okay or if you have a different suggestion for where this link should go.

**Comment by CloVasco** [comment]
@laurabenetech Thanks. I am ok with your suggestions.
@johnhbenetech let us know what you think and if you have comments/questions on this

**Comment by laurabenetech** [comment]
@CloVasco 
Menu item to view dismissed records:
<img width="279" alt="Menu item dismissed records" src="https://user-images.githubusercontent.com/48697326/57561021-57d11e80-733e-11e9-81cd-7e80da3d2099.png">

**Comment by NeilMBenetech** [comment]
We should also consider the ability to dismiss at the Service and Location level, independently of the Organization. But if the Organization is dismissed, all child records should be considered Dismissed too.

**Comment by johnhbenetech** [comment]
@NeilMBenetech if a previously dismissed record receives an update, should it still remain dismissed?

should we ever automatically redisplay a record? or the only way is to manually reinstate?

**Comment by NeilMBenetech** [comment]
@johnhbenetech  I'm thinking that upon any new change to the record, subsequent to the user dismissing it, it once again appears in the feed. For that reason we should consider making the onhover or button text itself more descriptive: "Dismiss until new changes" or something like that.

**Comment by awalkowiak** [comment]
@NeilMBenetech / @johnhbenetech / @laurabenetech 
What in case if there are multiple organization matches on card on home page? Should all be hidden (dismissed) at once?

![Selection_064](https://user-images.githubusercontent.com/12113312/62062065-99c87d00-b228-11e9-9d87-7ca5c4c6db07.png)

**Comment by johnhbenetech** [comment]
@awalkowiak Yes, the intent would be to hide that entire set of matches (essentially hide that whole card). Let me know if you come across an edge case we arent thinking of

**Comment by awalkowiak** [comment]
@johnhbenetech ok cool, no edge cases so far.

> the intent would be to hide that entire set of matches

But on the side by side view "hide" would be applied only to the currently visible match?

**Comment by bognasoldev89** [comment]
@johnhbenetech is viewing dissmissed records supposed to be possible only for admin users?

**Comment by johnhbenetech** [comment]
it should be for any user

**Comment by bognasoldev89** [comment]
I noticed a few things that need fixing:
1. Viewing dismissed records is possible only for admin users:

| | | |
| --- | --- | --- |
| ![308 bug 1](https://user-images.githubusercontent.com/51324797/62931037-e631c200-bdbd-11e9-82ff-95efae9f6fd7.png) | ![308 bug 2](https://user-images.githubusercontent.com/51324797/62931051-ef229380-bdbd-11e9-922d-f3073d80f627.png) | ![308 admin dropdown](https://user-images.githubusercontent.com/51324797/62931070-f6e23800-bdbd-11e9-9cbf-79a237d3b147.png) |

2. Column "User" on Hidden organization matches detail page should contain information which user dismissed the record. It is empty now:

![308 bug 3](https://user-images.githubusercontent.com/51324797/62931209-37da4c80-bdbe-11e9-8382-d925bfbb78bf.png)

3. After dismissing record on multirecord view the page should reload. At this moment user has to reload page manually. For now, instant effect is visible only after dismissing only when there is one match:

![308 multi record view](https://user-images.githubusercontent.com/51324797/62931524-cf3f9f80-bdbe-11e9-8905-ae549a2333fd.gif)

4. As I understand, dismissed record when updated should appear on the homepage again. It did not.

5. Tooltip appears only on the first record on the homepage:

| |
| --- |
| ![308 bug 4 1](https://user-images.githubusercontent.com/51324797/62931794-30677300-bdbf-11e9-9cd8-28aed20ffe3e.png) |
| ![308 bug 4 2](https://user-images.githubusercontent.com/51324797/62931806-378e8100-bdbf-11e9-8daa-de14bb9fbb8c.png) |

**Comment by bognasoldev89** [comment]
@johnhbenetech I have one more question. Is dismissing at Service and Location level in the scope of this task? It was mentioned but it is not clear if it was an idea to improve this feature.

**Comment by johnhbenetech** [comment]
No. That will be a separate task if we decide to do it

________________________________

**Comment by bognasoldev89** [comment]
I tested this issue again and all mistakes are fixed.
I noticed another issue though: is one regular user supposed to see and reinstate records dismissed by another regular user? It doesn't seem safe. Screenshot for example:

![hidden matches view regular user](https://user-images.githubusercontent.com/51324797/63854526-9fd38a00-c99d-11e9-825d-c3deb3abfc31.png)

@johnhbenetech what do you think? Leaving it for now, resolving in the scope of this ticket or maybe creating the new issue is the best solution?

**Comment by bognasoldev89** [comment]
Of course, I will!
And I'm closing this issue, as I said, mistakes are fixed and feature works well.
